tests/utils.ts

```diff
@@ -32,23 +37,69 @@ export function requireEnvVar(name: string): string {
   return value;
 }
 
-let tasksToRemove: number[] = [];
-export const queueTaskCleanup = (id: string) => {
-  if (id in tasksToRemove) {
-      return;
+// Add a workflow to the list of created workflows for cleanup, or removal later
+export const queueForRemoval = (
+  workflowIds: Map<string, boolean>,
+  workflowId: string
+) => {
+  if (workflowIds.has(workflowId)) {
+    return;
   }
-  tasksToRemove.push(id);
-}
 
-export const teardown = async (client, authKey): Promise<void> => {
-  await Promise.all(tasksToRemove.map(async (id) => {
-    try {
-      await client.deleteTask(id, { authKey });
-    } catch (e) {
-      // this is a cleanup during test, it isn't an issue if the delete failed.
-      console.log("cannot cleanup task id. please clean up manually", id, e)
-    }
-  }));
-
-  tasksToRemove = [];
-}
\ No newline at end of file
+  workflowIds.set(workflowId, false);
+};
+
+// Remove all workflows from the list of created workflows
+export const removeCreatedWorkflows = async (
+  client: Client,
+  authKey: string,
+  workflowIds: Map<string, boolean>
+): Promise<void> => {
+  await Promise.all(
+    Array.from(workflowIds.entries()).map(async ([workflowId, isDeleting]) => {
+      // Prevent re-entry with the same workflow id
+      if (isDeleting) {
```

Comment:
The only change is to add this flag, to prevent deletion of the same workflowId multiple times. It doesn’t break the tests but will print out console errors, and send unnecessary delete calls.